Add unit test for function bitmap_filename

rory-cd opened the pull request:
# Description

Adds unit test for the `bitmap_filename` function from _images.cpp_ (documentation [here](https://splashkit.io/api/graphics/#bitmap-filename)). The new test is named _"bitmap filename can be retrieved"_.

This test includes sections covering the following cases:
- Valid bitmap (loaded from file): Returns filename
- Null reference: Returns empty string
- Newly created bitmap: Returns empty string

**Note:** Although the function uses the term _filename_, it actually returns the full file path. So in the first section, the test extracts the file name from the full path before making its comparison. It may be worth renaming the function to `bitmap_filepath` and adding a new `bitmap_filename` to make this clearer for the end user, but that is beyond the scope of this update.

## Type of change

- [✓] Unit test addition

## How Has This Been Tested?

Built the test project as per [unit testing instructions](https://thoth-tech.netlify.app/products/splashkit/documentation/splashkit-expansion/01-unit-testing-guide/), running _skunit_tests_ with all tests passed. The returned strings were also manually verified by outputting each using `std::cout`.

## Testing Checklist

- [ ✓] Tested with skunit_tests

## Checklist

- [✓] My code follows the style guidelines of this project
- [✓] I have performed a self-review of my own code
- [✓] I have commented my code in hard-to-understand areas
- [   ] I have made corresponding changes to the documentation
- [✓] My changes generate no new warnings
- [   ] I have requested a review from ... on the Pull Request

rory-cd committed:
Add bitmap_filename unit test

222448082Ashen reviewed (commented):
# Description

This update adds a new unit test for the `bitmap_filename` function in `images.cpp`.
The test is named **"bitmap filename can be retrieved"** and checks three cases:

* A loaded bitmap returns the correct filename
* A null bitmap returns an empty string
* A newly created bitmap returns an empty string

The function currently returns a full file path.
For the comparison, the test takes only the last part of the path.
A rename to `bitmap_filepath` could help long-term clarity, but that is outside this change.

## Type of change

* [x] Unit test addition

## How Has This Been Tested?

The test suite was built and run using the normal unit test flow.
`skunit_tests` completed with all tests passing.
Output strings were also printed to confirm each value manually.

### Testing Checklist

* [x] Tested with `skunit_tests`

## Checklist

* [x] Code follows project style
* [x] Self-review completed
* [x] Comments added where helpful
* [ ] Documentation not required for this update
* [x] No new warnings
* [ ] Reviewer will be added when opening PR

## Files Modified

* Updated: `unit_test_bitmap.cpp` — added new sections testing `bitmap_filename`

## Additional Notes

No API changes or behaviour changes included.

rory-cd committed:
Merge remote-tracking branch 'upstream/main' into test/bitmap_filename

ralphweng2023 commented on coresdk/src/test/unit_tests/unit_test_bitmap.cpp:
```diff
@@ -134,3 +136,39 @@ TEST_CASE("bitmap bounding details can be retrieved", "[bitmap]")
+        // Extract filename
+        std::string filepath = bitmap_filename(bmp);
+        size_t idx = filepath.size() - expected_filename.size();
+        std::string filename = filepath.substr(idx, expected_filename.size());
```
###  Unsafe substring extraction

This code has a potential unsigned integer underflow if `filepath` is shorter than `expected_filename`:

```cpp
size_t idx = filepath.size() - expected_filename.size();  // Can underflow!
std::string filename = filepath.substr(idx, expected_filename.size());
```

I would suggest do it this way:
```
std::string filepath = bitmap_filename(bmp);
REQUIRE(filepath.size() >= expected_filename.size());
std::string filename = filepath.substr(filepath.size() - expected_filename.size());
REQUIRE(filename == expected_filename);
```

ralphweng2023 commented:
## Peer Review Summary

Thanks for this contribution! I've reviewed the PR against the SplashKit peer review checklist. Overall, the unit tests are well-structured and follow the existing conventions, but there are a couple of issues that need to be addressed before approval.

---

### ✅ **Code Quality**

- [x] **Repository**: Correct repository (splashkit-core)
- [x] **Readability**: Code is clear and follows Catch2 conventions
- [x] **Maintainability**: Tests are modular and follow existing test structure

### ✅ **Functionality**

- [x] **Correctness**: Tests cover the main behaviors of `bitmap_filename`
- [x] **Impact on Existing Functionality**: No breaking changes; only adds tests

### ✅ **Testing**

- [x] **Test Coverage**: Basic scenarios covered (loaded bitmap, null bitmap, created bitmap)
- [x] **Test Results**: Author reports 80 tests, 0 failures (100% pass)

### ⚠️ **Issues Found**

#### 1. **Safety Issue** (inline comment added)
The substring extraction in the first test case has a potential unsigned integer underflow. Please see my inline comment with suggested fix.


---

### **Decision: Request Changes** ⚠️

Please address the two inline comments before this can be approved. Once fixed, I'll be happy to approve!

Let me know if you have any questions about the suggested changes.

rory-cd committed:
Fix potential integer underflow as per review

Adds requirement ensuring the filepath being read exceeds or equals
the length of the expected filename. This removes the potential for
integer underflow.

rory-cd committed:
Merge remote-tracking branch 'upstream/main' into test/bitmap_filename

rory-cd commented:
Thanks @ralphweng2023, nice catch! I have updated to address the issue.

I've followed your suggestion, but used an `idx` variable, which hopefully makes it a little easier to read.

rory-cd committed:
Merge branch 'main' into test/bitmap_filename
